### PR TITLE
Breakdown trees

### PR DESCRIPTION
**Pull request by MeghaChakraborty**
Adding Spatial Containment Tree, Classifications Tree and Component index.

### CODE REVIEW

**Review by aruniverse — COMMENTED** [review]
would it be possible for you to lint and format thee files? ive only covered a 1/3 of the code so far

File: packages/tree-widget/src/components/rulesets/ClassificationSystems.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
                             ]
                         }
                     ],
-                    "arePolymorphic": false,
+                    "arePolymorphic": true,
```
[the remark this answers is not in the document]
Comment (diegopinate) [reply packages/tree-widget/src/components/rulesets/ClassificationSystems.json:17]: I don't see a problem with the specific "arePolymorphic" change you are highlighting here,  but as I talked with you yesterday, maybe @jason-crow or Heather would be best to review these rulesets?

[the remark this answers is not in the document]
Comment (jason-crow) [reply packages/tree-widget/src/components/rulesets/ClassificationSystems.json:17]: Well  it depends on if this needs to work polymorphically. @MeghaChakraborty is this an intentional fix. I know polymorphic rulesets will work slower, but there's certainly cases you want to use it. I'm not familiar enough with the Classifications Tree to know the right answer here

[the remark this answers is not in the document]
Comment (JValiunas) [reply packages/tree-widget/src/components/rulesets/ClassificationSystems.json:17]: arepolymorphic: false will be very rare in presentation rules, the non-pylomorphic approach can only be reliably used with final classes (and those are quite rare in the schemas i've seen so far). Classes ClassificationSystems are not final, so yes, this needs to be polymorphic.

so i'd turn this around by saying non-polymorphic rulesets will not be the right choice most of the times, but in certain rare cases you want it to be non-polymorphic (for example, final classes)

[the remark this answers is not in the document]
Comment (grigasp) [reply packages/tree-widget/src/components/rulesets/ClassificationSystems.json:17]: Note - `arePolymorphic` flag has no effect for final classes (even performance-wise). Generally `arePolymorphic: false` should be used in very specific (and probably rare) cases when you want to get instances of specified class but not its subclasses.

File: packages/tree-widget/src/components/trees/FunctionalityProviders/BuildingClipPlanesProvider.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  private async executeQuery(iModel: IModelConnection, query: string) {
+    const rows = [];
+    for await (const row of iModel.query(query)) rows.push(row);
+    return rows;
+  }
+
+  private async checkIsBuilding(iModel: IModelConnection, bldgId: string) {
+    let query = "";
+    query = `SELECT ECInstanceId as id FROM BuildingSpatial.Building WHERE ECInstanceId = ${bldgId}`;
```
Comment (aruniverse) [inline packages/tree-widget/src/components/trees/FunctionalityProviders/BuildingClipPlanesProvider.ts:30]: could make this a const, youre not redefining this

File: packages/tree-widget/src/components/trees/FunctionalityProviders/SpaceClipPlanesProvider.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    this.defaultHeight = defaultHeight;
+  }
+
+  public async performAction(node: TreeModelNode) {
+    await this.clipViewToSpace(node);
+  }
+
+  private async executeQuery(iModel: IModelConnection, query: string) {
+    const rows = [];
+    for await (const row of iModel.query(query)) rows.push(row);
```
Comment (aruniverse) [inline packages/tree-widget/src/components/trees/FunctionalityProviders/SpaceClipPlanesProvider.ts:27]: why not just return iModel.query(query)?

File: packages/tree-widget/src/components/trees/FunctionalityProviders/SpaceClipPlanesProvider.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    return rows;
+  }
+
+  private async checkIsSpace(iModel: IModelConnection, spaceId: string) {
+    let query = "";
+    query = `SELECT ECInstanceId as id FROM BuildingSpatial.Space WHERE ECInstanceId = ${spaceId}`;
```
Comment (aruniverse) [inline packages/tree-widget/src/components/trees/FunctionalityProviders/SpaceClipPlanesProvider.ts:33]: const

**Pull request closed without merging**